phone_proxy/lib/main.dart

```diff
@@ -1,149 +1,231 @@
+import 'package:blev/ble_central.dart';
+import 'package:blev/ble_peripheral.dart';
+import 'package:blev/ble_socket.dart';
+
 import 'package:flutter/material.dart';
+import 'package:logger/logger.dart';
 import 'package:permission_handler/permission_handler.dart';
 import 'package:socks5_proxy/socks_server.dart';
 
-List<String> lines = [];
+// The following three classes are examples of a basic flutter app setup.
+// Replace them with the implementation of your flutter app.
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+        title: 'Phone Proxy', home: MyHomePage(title: 'Phone Proxy'));
+  }
+}
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+class _MyHomePageState extends State<MyHomePage> {
+  _MyHomePageState() {
+    loadData();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          backgroundColor: Theme.of(context).colorScheme.inversePrimary,
+          title: Text(widget.title),
+        ));
+  }
+
+  Future<void> loadData() async {
+    while (true) {
+      await Future<void>.delayed(const Duration(seconds: 1));
+      setState(() {});
+    }
+  }
+}
+
+// `main` is an example of how your flutter app might call into `startBLESocksPhoneProxy`
+// Specifically, you will have to provide the values for `mobileDevice` and
+// `machineToManage`.
+void main() {
+  // `mobileDevice` should be stored somewhere in a mobile app. It should be
+  // unique to a mobile device such that `startBLESocksPhoneProxy` can:
+  // - Advertise `mobileDevice` as a readable characteristic
+  // - Write `mobileDevice` to a characteristic on the machine to manage
+  // - Allow L2CAP connections from that managed machine
+  //
+  // You may want to hardcode a unique `mobileDevice` value in each instance of
+  // your app.
+  var mobileDevice = 'd3e535a.viam.cloud';
 
-// This should be stored somewhere in a mobile app.
-var deviceName = 'd3e535ca.viam.cloud';
+  // `machineToManage` should be the machine name (FQDN) of the machine for
+  // which the mobile device is trying to proxy traffic. Assuming it is
+  // running the `socks-forwarder`, the managed machine should already be:
+  // - Advertising `machineToManage` as a readable characteristic
+  // - Adveristing a writable characteristic (encrypted) to which the mobile
+  //   device will need to write its `mobileDevice` value
+  // - Ready to establish an L2CAP connection for SOCKS forwarding once a value
+  //   is written to the above characteristic
+  //
+  // You may want to ask users to enter the `machineToManage` value as part
+  // of app setup.
+  var machineToManage = 'mac1.loc1.viam.cloud';
 
-var machineToManage = 'mac1.loc1.viam.cloud';
+  runZonedGuarded(
```

Comment:
`runZoned` -> `runZonedGuarded` due to some advice I read online and a deprecation warning for `runZoned`.